flake: Test_mustStartExtProc

*Description*:

https://github.com/envoyproxy/ai-gateway/actions/runs/17328278371/job/49197814824

```
--- FAIL: Test_mustStartExtProc (1.00s)
    run_test.go:224: 
        	Error Trace:	/Users/runner/work/ai-gateway/ai-gateway/cmd/aigw/run_test.go:224
        	Error:      	Received unexpected error:
        	            	external processor run error: failed to listen for health checks: listen tcp :1065: bind: address already in use
        	Test:       	Test_mustStartExtProc
```

i think we should allow health check port configuration, and use an ephemeral port in the tests

--- a/cmd/aigw/run_test.go
+++ b/cmd/aigw/run_test.go
@@ -207,15 +207,13 @@ func TestRunCmdContext_writeEnvoyResourcesAndRunExtProc(t *testing.T) {
 }
 
 func Test_mustStartExtProc(t *testing.T) {
-	ctx, cancel := context.WithCancel(t.Context())
-	t.Cleanup(cancel)
 	mockerr := errors.New("mock extproc error")
 	runCtx := &runCmdContext{
 		tmpdir:          t.TempDir(),
 		extProcLauncher: func(context.Context, []string, io.Writer) error { return mockerr },
 		stderrLogger:    slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{})),
 	}
-	done := runCtx.mustStartExtProc(ctx, filterapi.MustLoadDefaultConfig())
+	done := runCtx.mustStartExtProc(t.Context(), filterapi.MustLoadDefaultConfig())
 	require.ErrorIs(t, <-done, mockerr)
 }
 

